csrc/scheduler/greedy.cpp

```diff
@@ -705,22 +755,65 @@ class HeuristicsBuilder : private IterVisitor {
   }
 
   void handle(ArgsortOp* argsort) override {
-    addHeuristicsFor(ir_utils::getTvOutput(argsort), {argsort->dim()});
+    addHeuristicsFor(
+        ir_utils::getTvOutput(argsort),
+        ir_utils::getTvOutput(argsort)->getLogicalDomain(),
+        {argsort->dim()});
+  }
+
+  // TODO: Support batching
+  void handle(PadOp* pad) override {
+    setDefaultParameters(ir_utils::getTvOutput(pad));
   }
 
   void handle(ScanOp* scan) override {
-    addHeuristicsFor(ir_utils::getTvOutput(scan), {scan->dim()});
+    addHeuristicsFor(
+        ir_utils::getTvOutput(scan),
+        ir_utils::getTvOutput(scan)->getLogicalDomain(),
+        {scan->dim()});
+  }
+
+  void handle(ScatterOp* scatter) override {
+    // Need to have two sets of parameters: one for the logical domain
+    // of the index tensor and another for the logical domain of the
```

Comment:
super nitpick: the comment here says the first parameter is for `the logical domain of the index tensor`. which maps to the `initial loop domain of the output tensor` and that's what we have in the code below.

The comment we have sounded a bit strange to me, because it suggests that this is mapping it as a producer.